test/celeritas/field/FieldPropagator.test.cc

```diff
@@ -466,6 +466,26 @@ TEST_F(TwoBoxTest, gamma_exit)
     }
 }
 
+TEST_F(TwoBoxTest, electron_super_small_step)
+{
+    auto particle = this->init_particle(
+        this->particle()->find(pdg::electron()), MevEnergy{2});
+    UniformZField field(1 * units::tesla);
+    FieldDriverOptions driver_options;
+    for (real_type delta : {1e-14, 1e-8, 1e-2, 0.1})
```

Comment:
It seems that 1e-14 and 1-e8 are practically too small.

Reply:
We do see a fair number of these very small steps in the demo loop -- they seem to be mainly range-limited steps for charged particles with extremely low energies, but there are also some cases where the remaining MFP is super small, leading to an extremely small interaction length.

Reply:
Yep. It's rare but it seems to happen, therefore we need to make sure it works. (Or we need an error recovery mechanism, but it's better to just fix it if we can.)

Reply:
As long as the field propagation concerns, we should not apply the field integration for the case when the step length is below a certain threshold, currently minimum_step = 1.0e-5 * units::millimeter.  As it will die after a small step, which not affect any physics result.

Reply:
If below the threshold then, should we just do a linear trace? We still want to physically move the particle those small distances, and we need to do the boundary intersections even if the distance is small.

Reply:
Exactly.  The primary purpose of the field integration is to evaluate the curvature of track (small directional change at the end of track with an extra very small step length does not practically affect anything and the linear propagator should be good enough).  Anyway, let's merge this for now once the ci test is passed, but we may revisit self-consistent criteria when the field or linear propagator are applied.